RyuSocks/Packets/IAuthenticationPacket.cs

```diff
@@ -0,0 +1,7 @@
+namespace RyuSocks.Packets
+{
+    public interface IAuthenticationPacket : IPacket
```

Comment:
This should either be removed (if only `Version` is shared between all auth packets) or changed to an abstract class, like the other packet classes that are not meant to be used directly (see: `CommandPacket`).